shap/plots/_heatmap.py

```diff
@@ -178,3 +182,5 @@ def heatmap(shap_values, instance_order=Explanation.hclust(), feature_values=Exp
 
     if show:
         pl.show()
+
+    return ax
```

Comment:
The bar plot currently has this behaviour, only returning `ax` if `show=False`:

```python
    if show:
        pl.show()
    else:
        return ax
```

Any thoughts @CloseChoice on which convention is better? I'm happy with either, but we should probably be consistent. I think we should also be mindful to minimise breaking changes wherever possible.

Reply:
Maybe I can add my thoughts. When I use shap in jupyter notebooks/lab, the default is to automatically show plots created by Matplotlib/Seaborn. For me,  the pl.show() approach without returning an ax was irritating because it always took a bit of extra code to get the figure for saving. 

In contrast, when the ax is returned, I can get both the ax/fig for saving and I can see the plot.

Reply:
Isn't the figure shown twice in notebooks if we do 
```python
if show:
    pl.show()
return ax
```
? If so, then I'd think an additional `else` would be good.
If not, then let's just avoid the additional `else`

Reply:
@CloseChoice Good question. I tried it with the following code and it does not show the figure twice:

```python
from matplotlib import pyplot as pl
import numpy as np

# define test function
def simple_plot(ax=None, show=True):
    
    if ax is None:
        pl.gcf().set_size_inches(6,8)
        ax = pl.gca()
    data = np.arange(0,100)
    ax.plot(data)
    
    if show:
        pl.show()
    
    return ax

# test without passing an ax

testfig = simple_plot(show=True)

# test with passing an ax

fig,ax = pl.subplots()
test_fig = simple_plot(ax=ax)
```

Note that the function returns an ax and not a figure. If the figure should be accessed/saved it needs to be retrieved using the `.get_figure()` method.

Reply:
Nice, thanks for checking. Then I would suggest to go with the 
```python
if show:
    pl.show()
return ax
```
solution